### PR TITLE
React hooks

### PR DESCRIPTION
**Pull request by ecstaciEva**
Removed unnecessary callback setState, and fixed the way of using setState.

### CODE REVIEW

File: src/Components/AddTodoForm.tsx [hunk 1]
```diff
@@ -32,7 +33,7 @@ export const AddTodoForm: React.FC<Props> = ({ addTodo }) => {
   const classes = useStyles();
 
   return (
-    <form className={classes.addTodoForm}>
+    <FormControl className={classes.addTodoForm}>
```
Comment (ecstaciEva) [inline src/Components/AddTodoForm.tsx:36]: I used FormControl but with 2 inputs inside it ( input and submit),
so I changed the element to form according to [the previous code-review](https://github.com/ecstaciEva/todo-list-typescript/pull/2#discussion_r637834338).

I changed it back to FormControl and remove the type="submit" from the Button this time,
but I'm not sure whether it is the right component I should be using.

Comment (laurenchen0631) [reply src/Components/AddTodoForm.tsx:36]: You can't use `<form>` since I only allow material-ui in this project.

Comment (laurenchen0631) [reply src/Components/AddTodoForm.tsx:36]: But you can use `<Box component="form">` to create a form element.

File: src/Components/App.tsx [hunk 2]
```diff
@@ -34,65 +44,73 @@ const App: React.FC = () => {
...
+          }
+          return item;
+        });
+        return newTodos;
       });
-<<<<<<< HEAD
-      setTodos(newTodos);
-=======
-      setTodos(() => newTodos);
->>>>>>> auto-format
```
Comment (laurenchen0631) [inline src/Components/App.tsx:50]: Do not commit any conflicted code to your project.

File: src/Components/App.tsx [hunk 3]
```diff
@@ -34,65 +44,73 @@ const App: React.FC = () => {
...
+          if (todo.id === item.id) {
+            return {
+              ...item,
+              isComplete: !item.isComplete,
+            };
+          }
+          return item;
+        });
+        return newTodos;
       });
```
Comment (laurenchen0631) [inline src/Components/App.tsx:58]: correct.

File: src/Components/TodoList.tsx [hunk 4]
```diff
@@ -37,13 +37,12 @@ export const TodoList: React.FC<Props> = ({
...
-    todos.filter((todo) => {
-      todo.isComplete ? completedTodos.push(todo) : activeTodos.push(todo);
-    });
-  }, [todos]);
+  const todoCategory = (iscomplete: boolean): Todo[] => {
+    const categoryList = todos.filter((todo) => todo.isComplete === iscomplete);
+    return categoryList;
+  };
+  const activeTodos = todoCategory(false);
+  const completedTodos = todoCategory(true);
```
Comment (laurenchen0631) [inline src/Components/TodoList.tsx:45]: The right way:
`const activeTodos =  useMemo(() => todos.filter((todo) => !todo.isComplete, [todos])`.

And, you're right about that the performance won't change for this design.
But, it just a practice to make sure your skill of react and typescript

**Review by laurenchen0631 — APPROVED** [review]

**Pull request merged**